parser/src/ast/operation.rs

```diff
@@ -1,4 +1,13 @@
+use enum_assoc::Assoc;
+use lexer::token::TokenType;
+
 use crate::ast::Expr;
+use crate::ast::operation::BinaryOperator::*;
+
+pub const ARITHMETICS: &[BinaryOperator] = &[Plus, Minus, Times, Divide, Modulo];
+pub const COMPARISONS: &[BinaryOperator] = &[EqualEqual, NotEqual, Less, LessEqual, Greater, GreaterEqual];
+pub const LOGICALS: &[BinaryOperator] = &[And, Or];
```

Comment:
These constants don't seem to be used anywhere. If they could be used elsewhere, it would be better to have a method on `TokenType` for each group.

Reply:
done